src/common/telemetry/collect-telemetry/collect-telemetry.ts

```diff
@@ -171,7 +172,14 @@ export const collectTelemetry = async (project: NodeProject & IWithTelemetryRepo
   try {
     const body = JSON.stringify(payload)
     project.logger.info('Collected telemetry:', body)
-    const {status, statusText} = await fetch(project.telemetryReportUrl!, {method: 'post', body})
+    const headers: Record<string, string> = {}
+    const authHeaderName = project.telemetryAuthHeader
+
+    if (authHeaderName) {
+      headers[authHeaderName] = process.env[telemetryAuthToken]!
```

Comment:
@quesabe would this make naming more consistent?
```suggestion
      headers[reportTargetAuthHeaderName] = process.env[reportTargetAuthToken]!
```

Also let's probably rename `project.telemetryReportUrl` to `project.reportTargetUrl`.

Reply:
@gvidon Updated the option names.

BTW, since we added more options shouldn't we move them into an object instead of residing them all on the root of the options. E.g.:
```typescript
export interface TelemetryOptions {
  /**
   * Endpoint URL to send telemetry data to.
   */
  readonly reportTargetUrl?: string

  /**
   * Authorization header name for telemetry
   */
  readonly reportTargetAuthHeaderName?: string

  /**
   * The name of env var to extract header value from.
   * The value is expected to be stored in a CI secret.
   */
  readonly reportTargetAuthTokenVar?: string
}

export interface WithTelemetry {
  /**
   * Enable template usage telemetry.
   * Collects the data on the template package version, connected git remotes,
   * applied escape hatches, configured GitHub workflows.
   *
   * @default false
   */
  readonly isTelemetryEnabled?: boolean

  /**
   * Configuration options for telemetry
   */
  readonly telemetryOptions?: TelemetryOptions
}
```

Reply:
@quesabe yeah, an object is a good idea. Should we add it per this PR?